amulet/level/formats/anvil_forge_world.py

```diff
@@ -40,14 +41,27 @@ def game_version_string(self) -> str:
             return f"Java Forge Unknown Version"
 
     def _register_modded_dimensions(self):
-        for path, directories, files in os.walk(os.path.join(self.path, "dimensions")):
-            if set(("data", "entities", "poi", "region")).issubset(set(directories)):
-                dimension_path_split = path.split(os.sep)
-                dimensions_directory_index = dimension_path_split.index("dimensions")
-                dimension_name = f"{dimension_path_split[dimensions_directory_index + 1]}:{'/'.join(dimension_path_split[dimensions_directory_index + 2:])}"
-                self._register_dimension(
-                    os.path.dirname(os.path.relpath(path, self.path)), dimension_name
+        for region_path in glob.glob(
+            os.path.join(glob.escape(self.path), "dimensions", "*", "**", "region"),
+            recursive=True,
+        ):
+            dim_path = os.path.dirname(region_path)
+            child_dir_names = set(
+                map(
+                    os.path.basename,
+                    filter(
+                        os.path.isdir,
+                        map(lambda d: os.path.join(dim_path, d), os.listdir(dim_path)),
+                    ),
                 )
+            )
```

Comment:
I had missed that listdir returns base names not the full path.
This might be better as a comprehension so that you don't need to join and then split the path.
```py
child_dir_names = set(basename for basename in os.listdir(dim_path) if os.path.isdir(os.path.join(dim_path, basename))
```